new-lamassu-admin/src/pages/Services/Services.js

```diff
@@ -56,6 +57,47 @@ const Services = () => {
     }))(faceElements)
   }
 
+  const getElements = ({ code, elements }) => {
+    return R.compose(
+      R.map(elem => {
+        return elem.component === SecretInput
+          ? R.assoc(
+              'inputProps',
+              {
+                isPasswordFilled:
+                  !R.isNil(accounts[code]) &&
+                  !R.isNil(R.path([elem.code], accounts[code]))
+              },
+              elem
+            )
+          : R.identity(elem)
+      })
+    )(elements)
+  }
```

Comment:
This is overall pretty hard to parse.

You can do

```
if (elem.component !== SecretInput) return elem
return R.assoc('inputProps',
```

instead of the ternary if.

Reply:
Also looks like R.assoc does a shallow clone, can't we have nested objects here?